animation_nodes/nodes/mesh/set_vertex_weights.py

```diff
@@ -0,0 +1,157 @@
+    ("INDEX", "Index", "Set weight of a specific vertex", "NONE", 1)
+]
+
+groupIdentifierTypeItems = [
+    ("INDEX", "Index", "Get vertex group based on the index", "NONE", 0),
+    ("NAME", "Name", "Get vertex group based on the name", "NONE", 1)
+]
+
+groupNotFoundMessage = "group not found"
+noMeshMessage = "no mesh object"
+editMeshMessage = "object is not in object mode"
+
+class VertexWeights(bpy.types.Node, AnimationNode):
+    bl_idname = "an_SetVertexWeights"
+    bl_label = "Set Vertex Weights"
+    errorHandlingType = "EXCEPTION"
+
+    mode: EnumProperty(name = "Mode", default = "ALL",
+        items = modeItems, update = AnimationNode.refresh)
+
+    groupIdentifierType: EnumProperty(name = "Group Identifier Type", default = "INDEX",
+        items = groupIdentifierTypeItems, update = AnimationNode.refresh)
+
+    useIndexList: VectorizedSocket.newProperty()
+    useWeightsList: VectorizedSocket.newProperty()
+ 
+    def create(self):
+        self.newInput("Object", "Object", "object", defaultDrawType = "PROPERTY_ONLY")
+
+        if self.groupIdentifierType == "INDEX":
+            self.newInput("Integer", "Group Index", "groupIndex")
+        elif self.groupIdentifierType == "NAME":
+            self.newInput("Text", "Name", "groupName")
+
+        if self.mode == "INDEX":
+            self.newInput(VectorizedSocket("Integer", "useIndexList",
+                ("Index", "index"), ("Indices", "indices")))
+            self.newInput(VectorizedSocket("Float", "useWeightsList",
+                ("Weight", "weight"), ("Weights", "weights")))
+        elif self.mode == "ALL":
+            self.newInput(VectorizedSocket("Float", "useWeightsList",
+                ("Weight", "weight"), ("Weights", "weights")))
+        
+        self.newOutput("Object", "Object", "object")
+
+    def draw(self, layout):
+        layout.prop(self, "mode", text = "")
+
+    def drawAdvanced(self, layout):
+        layout.prop(self, "groupIdentifierType", text = "Type")
+
+    def getExecutionFunctionName(self):
+        if self.mode == "INDEX":
+            if self.useIndexList:
+                if self.useWeightsList:
+                    return "execute_IndicesList"
+                else:
+                    return "execute_IndicesSingle"
+            else:
+                return "execute_Index"
+        elif self.mode == "ALL":
+            if self.useWeightsList:
+                return "execute_AllList"
+            else:
+                return "execute_AllSingle"
+
+    def execute_Index(self, object, identifier, index, weight):
+        if object is None: return
+        if object.type != "MESH": 
+            self.raiseErrorMessage(noMeshMessage)
+        if object.mode == "EDIT":
+            self.raiseErrorMessage(editMeshMessage)
+
+        vertexGroup = self.getVertexGroup(object, identifier)
+        if vertexGroup is None:
+            self.raiseErrorMessage(groupNotFoundMessage)
```

Comment:
Since those lines are shared between all functions, can we move them to their own function?

Reply:
Yes, it is possible. Something like this,
```
    def execute_AllList(self, object, identifier, weights):
        if object is None: return
        vertexGroup = self.check(object, identifier)

        weights = VirtualDoubleList.create(weights, 0).materialize(len(object.data.vertices))
        for i, weight in enumerate(weights):
            vertexGroup.add([i], weight, "REPLACE") 
        object.data.update()
        return object            

    def check(self, object, identifier):
        if object.type != "MESH": 
            self.raiseErrorMessage(noMeshMessage)
        if object.mode == "EDIT":
            self.raiseErrorMessage(editMeshMessage)

        vertexGroup = self.getVertexGroup(object, identifier)
        if vertexGroup is None:
            self.raiseErrorMessage(groupNotFoundMessage)
        return vertexGroup

```